.github/workflows/ci.yml

```diff
@@ -0,0 +1,52 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request: {}
+
+concurrency:
+  # Group concurrency on workflow, then:
+  # - Is merge run? Group `push` events
+  # - Is pull request? Group `pull_request` events
+  group: >-
+    ${{ github.workflow }}-${{
+      github.event_name == 'push'
+        && github.base_ref
+        || github.head_ref
+    }}
+  # Run merge runs in sequence. Cancel in progress pull request runs.
+  cancel-in-progress: ${{ github.event_name != 'push' }}
+
+env:
+  NODE_OPTIONS: --max-old-space-size=6144
+
+jobs:
+  main:
+    name: Nx Cloud - Main Job
+    uses: nrwl/ci/.github/workflows/nx-cloud-main.yml@v0.4
+    with:
+      parallel-commands: |
+        yarn nx-cloud record -- yarn nx workspace-lint
+        yarn nx-cloud record -- yarn nx format:check
+      parallel-commands-on-agents: |
+        yarn nx affected --target=lint --parallel=2
+        yarn nx affected --target=test --parallel=2 --ci --code-coverage
+        yarn nx affected --target=build --parallel=2
+        yarn nx affected --target=e2e --parallel=1
+
+  agents:
```

Comment:
This is only for setting up things, right?

- Setting the agents
- Installing the packages
- Loading the cache

But we still need to run the necessary jobs like lint, test, build, etc right?

Reply:
Look 2 lines above 😄

This workflow is mostly generated by Nx. Let me add explanatory comments.

Reply:
Yeah, yeah now it is there :) this is an old comment!

Reply:
It was always there, Nacho 😁 I saw you adding this comment in your livestream. I don't know how you managed to overlook the lines above 😄